src/main/java/frc/robot/commands/LedBounceCommand.java

```diff
@@ -0,0 +1,123 @@
+
+package frc.robot.commands;
+
+import frc.robot.subsystems.LedSubsystem;
+import edu.wpi.first.wpilibj2.command.Command;
+
+/** An example command that uses an example subsystem. */
+public class LedBounceCommand extends Command {
+
+    private final LedSubsystem m_ledSubsystem;
+
+    private static final int[] m_primaryBounceColor    = {255, 0, 0 };
+    private static final int[] m_shadowBounceColor     = {30, 0, 0};
+    private static final int[] m_backgroundBounceColor = {20, 20, 20};
+
+    private static final int m_paceFactor              = 4;     // the execute routines runs every 20ms so this is a factor to slow us down
+    private static int       m_paceCount;
+    private static int       m_bounceIndex;
+    private static int       m_bounceDirection;
+
+
+    public LedBounceCommand(LedSubsystem subsystem) {
+        m_ledSubsystem = subsystem;
+        // Use addRequirements() here to declare subsystem dependencies.
+        addRequirements(subsystem);
+    }
+
+
+
+    // Called when the command is initially scheduled.
+    @Override
+    public void initialize() {
+
+        m_paceCount       = m_paceFactor;
+        m_bounceIndex     = 0;
+        m_bounceDirection = 1;
+
+        m_ledSubsystem.setAllLedsRGB(m_backgroundBounceColor[0],
+                                     m_backgroundBounceColor[1],
+                                     m_backgroundBounceColor[2]);
+
+        m_ledSubsystem.setLedRGB(0, m_shadowBounceColor[0],
+                                    m_shadowBounceColor[1],
+                                    m_shadowBounceColor[2]);
+                                
+        m_ledSubsystem.setLedRGB(1, m_primaryBounceColor[0],
+                                    m_primaryBounceColor[1],
+                                    m_primaryBounceColor[2]);
+                                
+        m_ledSubsystem.setLedRGB(2, m_shadowBounceColor[0],
+                                    m_shadowBounceColor[1],
+                                    m_shadowBounceColor[2]);
+    }
+
+
+
+    // Called every time the scheduler runs while the command is scheduled.
+    @Override
+    public void execute() {
+
+        // slow us down
+
+        m_paceCount--;
+
+        if (m_paceCount > 0 ) {
+            return;                     // don't do anyting - yet...
+        }
+
```

Comment:
The above serves as a loop, returning every single time it iterates. This causes the command to slow down because quiting the command out before restarting it takes a longer time to process, right?

Reply:
The reason for that check is to slow the pace of the LED changes.  The execute routine gets called at 50Hz or every 20ms.  If we just moved the LED pattern along every time that execute() got called, they would move very quickly.  What the code is doing instead is to move the LEDs only every m_paceCount entries into execute().  Most of the time it just returns and leaves the LEDs unchanged but every m_paceFactor times into execute() is the only time it moves the LED pattern.

Changing the value of m_paceCount changes how often the LED pattern actually moves.  In this case, we only move the LED every 4th time through so the LED pattern only moves every 80ms.

Changing m_paceCount controls the ratio of the LED movement to the frequency the execute() routine is being called.